<a href="https://colab.research.google.com/github/Vladimir2303850/MyAnalyticsProjects/blob/main/%D0%90%D0%BD%D0%B0%D0%BB%D0%B8%D0%B7_%D0%B4%D0%B0%D0%BD%D0%BD%D1%8B%D1%85_%D1%81%D1%82%D0%B0%D1%82%D0%B8%D1%81%D1%82%D0%B8%D1%87%D0%B5%D1%81%D0%BA%D0%B8%D0%B5_%D0%BC%D0%B5%D1%82%D0%BE%D0%B4%D1%8B.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **Задачи**

Определение математического ожидания для ключевых параметров (год выпуска, объем двигателя, пробег, цена и др.) для выявления средних характеристик автомобилей.

Вычисление стандартных отклонений и дисперсий для оценки разброса данных и вариативности параметров.

Расчет ковариации для изучения связей между такими параметрами, как год выпуска, пробег, количество владельцев и цена.

Интерпретация знака и величины ковариации для определения характера зависимостей (положительная/отрицательная).

Выявление ключевых факторов, оказывающих наибольшее влияние на цену.

Анализ слабых связей (год выпуска, объем двигателя) и их возможного вклада в стоимость.

Сравнение математического ожидания, медианы и стандартного отклонения для оценки распределения данных.

Формулирование выводов о типичных характеристиках автомобилей в выборке и их рыночной стоимости.

Определение значимых параметров для прогнозирования цены автомобилей.

Оценка надежности данных и их пригодности для принятия решений.

In [ ]:
import pandas as pd
import numpy as np

In [ ]:
df = pd.read_excel('data.xlsx')
df.head()

Марка  Модель   Год  Объем_двигателя Тип_топлива Коробка_передач  \
0         Kia     Rio  2020              4.2      Дизель        Механика   
1   Chevrolet  Malibu  2012              2.0      Гибрид         Автомат   
2    Mercedes     GLA  2020              4.2      Дизель         Автомат   
3        Audi      Q5  2023              2.0     Электро        Механика   
4  Volkswagen    Golf  2003              2.6      Гибрид     Полуавтомат   

   Пробег  Двери  Количество_владельцев   Цена  
0  289944      3                      5   8501  
1    5356      2                      3  12092  
2  231440      4                      2  11171  
3  160971      2                      1  11780  
4  286618      3                      3   2867

**Вычисление математического ожидания**

In [ ]:
numeric_data = df.select_dtypes(include=np.number)
mean_values = numeric_data.mean()
mean_values

Год                        2010.069767
Объем_двигателя               3.027907
Пробег                   130935.116279
Двери                         3.116279
Количество_владельцев         2.813953
Цена                       9029.674419
dtype: float64

Математическое ожидание года выпуска автомобилей — около 2010 года, что
указывает на их относительную новизну. Средний объем двигателя — 3.03 литра, что
характерно для автомобилей среднего и большого класса. Средний пробег — около 130 935
км, что говорит об активном использовании. В среднем у автомобилей 3 двери, а количество
владельцев — примерно 2.81, что может свидетельствовать о нескольких перепродажах.
Средняя цена — около 9029.67 единиц, что указывает на доступность. В целом, данные
показывают преобладание автомобилей среднего возраста с умеренным пробегом и средней
ценой.

**Вычисление стандартного отклонения**

In [ ]:
std_dev_values = numeric_data.std(ddof=1)
std_dev_values

Год                          6.822216
Объем_двигателя              1.126166
Пробег                   82325.435172
Двери                        1.117167
Количество_владельцев        1.384537
Цена                      2953.229360
dtype: float64

На основе представленных данных, где использована функция для вычисления
стандартного отклонения, можно сделать следующие выводы. Стандартное отклонение
года выпуска автомобилей составляет примерно 6.82 года, что указывает на умеренный
разброс в возрасте автомобилей. Стандартное отклонение объема двигателя — 1.13 литра,
что говорит о некоторой вариативности в размерах двигателей. Пробег автомобилей имеет
стандартное отклонение около 82 325 км, что свидетельствует о значительном разбросе в
пробегах. Стандартное отклонение количества дверей — 1.12, что указывает на
разнообразие типов кузовов. Количество владельцев имеет стандартное отклонение 1.38,
что может говорить о различной истории владения автомобилями. Стандартное отклонение
цены — 2953.23 единицы, что указывает на умеренный разброс в ценах. В целом, данные
показывают, что автомобили в наборе данных имеют значительную вариативность по
пробегу и умеренную — по остальным параметрам.

**Вычисление медианных значений**

In [ ]:
variance_values = numeric_data.var(ddof=1)
variance_values

Год                      4.654264e+01
Объем_двигателя          1.268250e+00
Пробег                   6.777477e+09
Двери                    1.248062e+00
Количество_владельцев    1.916944e+00
Цена                     8.721564e+06
dtype: float64

На основе представленных данных, где использована функция для вычисления
дисперсии, можно сделать следующие выводы. Дисперсия года выпуска автомобилей
составляет примерно 46.54, что указывает на умеренный разброс в возрасте автомобилей.
Дисперсия объема двигателя — 1.27, что говорит о некоторой вариативности в размерах
двигателей. Дисперсия пробега автомобилей составляет около 6.78 миллиардов км², что
свидетельствует о значительном разбросе в пробегах. Дисперсия количества дверей — 1.25,
что указывает на разнообразие типов кузовов. Дисперсия количества владельцев — 1.92,
что может говорить о различной истории владения автомобилями. Дисперсия цены
составляет около 8.72 миллиона единиц, что указывает на умеренный разброс в ценах. В
целом, данные показывают, что автомобили в наборе данных имеют значительную
вариативность по пробегу и умеренную — по остальным параметрам.

**Вычисление коэффициента ковариации**

In [ ]:
covariance_values = numeric_data.cov(ddof=1)
covariance_values

Год  Объем_двигателя        Пробег  \
Год                       46.542636        -0.775803  3.721940e+04   
Объем_двигателя           -0.775803         1.268250 -4.142313e+03   
Пробег                 37219.396456     -4142.312846  6.777477e+09   
Двери                      0.920266        -0.108084  1.476556e+04   
Количество_владельцев     -0.915282         0.183887  9.354927e+03   
Цена                   13383.023256       985.064064 -1.297732e+08   

                              Двери  Количество_владельцев          Цена  
Год                        0.920266              -0.915282  1.338302e+04  
Объем_двигателя           -0.108084               0.183887  9.850641e+02  
Пробег                 14765.557586            9354.926910 -1.297732e+08  
Двери                      1.248062               0.188815  2.179435e+02  
Количество_владельцев      0.188815               1.916944 -3.491573e+02  
Цена                     217.943522            -349.157254  8.721564e+06

На основе представленных данных, где вычислена ковариация для нескольких
столбцов датафрейма, можно сделать следующие выводы. Ковариация между годом
выпуска и пробегом составляет 3.72e+04, что указывает на слабую положительную связь:
более новые автомобили могут иметь меньший пробег. Ковариация между пробегом и
ценой равна -1.30e+08, что свидетельствует о сильной отрицательной связи: автомобили с
большим пробегом, как правило, дешевле. Ковариация между количеством владельцев и
ценой составляет -349.16, что также указывает на отрицательную связь: автомобили с
большим количеством владельцев могут быть дешевле. Ковариация между объемом
двигателя и ценой равна 985.06, что говорит о слабой положительной связи: автомобили с
большим объемом двигателя могут быть дороже. В целом, данные показывают, что пробег
и количество владельцев имеют значительное влияние на цену, в то время как год выпуска
и объем двигателя оказывают меньшее воздействие.



> **Вывод**

> *На основе анализа данных, включающего вычисление средних значений,
стандартных отклонений, дисперсий и ковариаций, можно сделать следующие общие
выводы. Автомобили в наборе данных в среднем выпущены около 2010 года, с умеренным
пробегом около 130 935 км и средней ценой примерно 9029.67 единиц. Объем двигателя в
среднем составляет 3.03 литра, а количество дверей — около 3. Стандартные отклонения и
дисперсии указывают на значительный разброс в пробеге и умеренный — в остальных
параметрах.
Ковариационный анализ выявил, что пробег и количество владельцев имеют сильное
отрицательное влияние на цену: автомобили с большим пробегом и большим количеством
владельцев, как правило, дешевле. Объем двигателя и год выпуска показывают слабую
положительную связь с ценой, что может указывать на то, что более новые автомобили и
автомобили с большим объемом двигателя могут быть дороже.
В целом, данные показывают, что пробег и количество владельцев являются
ключевыми факторами, влияющими на цену автомобилей, в то время как год выпуска и
объем двигателя оказывают меньшее, но все же заметное влияние*

